lib/jekyll/filters/url_filters.rb

```diff
@@ -51,15 +55,21 @@ def strip_index(input)
       private
 
       def sanitized_baseurl
-        site = @context.registers[:site]
-        site.config["baseurl"].to_s.chomp("/")
+        @baseurl ||= begin
```

Comment:
since this is a `Module` that is used as a mixin within `Jekyll::Filters`, would using a class instance variable have any unforeseen effect..?